File: server/models/player.js
const mongoose = require("mongoose");

const playerSchema = new mongoose.Schema({
    nickname: {
        type: String,
        trim: true,
    },
    socketId: {
        type: String,
    },
    points: {
        type: Number,
        default: 0,
    },
    playerType: {
        require: true,
        type: String,
    }
});

module.exports = playerSchema;

File: lib/models/player.dart
class Player {
  final String nickname;
  final String playerType;
  final String socketId;
  final double points;

  Player({
    required this.nickname,
    required this.playerType,
    required this.socketId,
    required this.points,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'nickname': nickname,
      'playerType': playerType,
      'socketId': socketId,
      'points': points,
    };
  }

  factory Player.fromMap(Map<String, dynamic> map) {
    return Player(
      nickname: map['nickname'] ?? '',
      socketId: map['socketID'] ?? '',
      points: map['points']?.toDouble() ?? 0.0,
      playerType: map['playerType'] ?? '',
    );
  }
}
